packages/lesswrong/lib/collections/dialogueMatchPreferences/schema.ts

```diff
@@ -0,0 +1,94 @@
+import SimpleSchema from "simpl-schema";
+import {schemaDefaultValue} from "../../collectionUtils";
+import { accessFilterSingle, foreignKeyField, resolverOnlyField } from "../../utils/schemaUtils";
+import DialogueChecks from "../dialogueChecks/collection";
+
+export const SYNC_PREFERENCE_VALUES = ['Yes', 'Meh', 'No'] as const;
+export type SyncPreference = typeof SYNC_PREFERENCE_VALUES[number];
+
+const topicPreferenceSchema = new SimpleSchema({
+  text: {
+    type: String,
+  },
+  preference: {
+    type: String,
+    allowedValues: ['Yes', 'No']
+  },
+  commentSourceId: {
+    type: String,
+    optional: true,
+    nullable: true,
+  }
+});
+
+const schema: SchemaType<DbDialogueMatchPreference> = {
+  dialogueCheckId: {
+    ...foreignKeyField({
+      collectionName: 'DialogueChecks',
+      type: 'DialogueCheck',
+      idFieldName: 'dialogueCheckId',
+      resolverName: 'dialogueCheck',
+      nullable: true
+    }),
+    nullable: false,
+    hidden: true,
+    canCreate: ['members', 'admins'],
+    canRead: ['members', 'admins'],
+    canUpdate: ['members', 'admins'],
+  },
+  topicPreferences: {
+    type: Array,
+    nullable: false,
+    canCreate: ['members', 'admins'],
+    canRead: ['members', 'admins'],
+    canUpdate: ['members', 'admins'],
```

Comment:
I'm pretty sure @Discordius intended to give this a default value of an empty array and am surprised nothing is breaking without that.